net2/BroDetect.js

```diff
@@ -1191,6 +1192,14 @@ module.exports = class {
                                 log.debug("HTTP:Save:Agent", host, o);
                             }
                         });
+                        let ukey = "user_agent:" + host + ":" + dst + ":" + dstPort;
+                        rclient.set(ukey, obj.user_agent, (err, response) => {
+                            if (err != null) {
+                                log.error("USER_AGENT:Save:Error", err, obj.user_agent);
+                            } else {
+                                rclient.expire(ukey, 3600); // a much shorter expiration since this is used to enrich alarm data
```

Comment:
you might need to extend this, since abnormal upload query window is 4 hours.

![image](https://user-images.githubusercontent.com/108800/43813176-13c1a8ce-9af7-11e8-9826-62b577c367b7.png)